File: JAVA/S9/L1/stack.java
class s{
    int size;
    int top=-1;
    int[]arr=new int[10];
    int isEmpty()
    {
        if(top==-1)return 1;
        return 0;
    }
    int isFull()
    {
        if(top==9)return 1;
        return 0;
    }
    void push(int data)
    {
        if(isFull()==1)return;
        arr[++top]=data;

    }
    int pop()
    {
        if(isEmpty()==1)return -1;
        return arr[top--];
        

    }
    void display()
    {
        for(int i=0;i<=top;i++)
        {
            System.out.print(arr[i]+" ");
        }
    }
}
public class stack
{
    public static void main(String[]args)
    {
        // int size=10;
        s st=new s();
        st.push(78);
        st.push(272);
        st.display();
    }
}

File: C/S1_BASIC/L1_BASIC/adt/stack/static/stack.c
//using array
#include<stdio.h>
#define size 2
int top=-1;//initialise top with -1
int arr[size];

//checking is stack is empty
int isEmpty(){
    return top==-1;
}

//checking is stack is full
int isFull(){
    return top==size-1;
}

//push element into stack
void push(int data){
    if(isFull()){
        printf("stack is full,can not push element\n");
        return ;
    }
    arr[++top]=data;
}
//remove from stack
int pop(){
    if(isEmpty()){
        printf("stack is empty,can not remove element\n");
        return -1;
    }
    return arr[top--];
}
//returns top element
int peek(){
    if(isEmpty())return -1;
    return arr[top];
}

//print elements
void display(){
    if(isEmpty()){
        printf("stack is empty,can not display element\n");
        return;
    }
    for(int i=0;i<=top;i++){
        printf("%d ",arr[i]);
    }
}
int main(){
    push(56);
    push(43);
    push(89);
    display();
    return 0;
}